DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -67,39 +66,65 @@ function Set-TargetResource
         [string]
         $Params,
         [ValidateNotNullOrEmpty()]
-        [string]
+        [string[]]
         $Version,
         [string]
         $Source,
-        [String]
+        [string]
         $chocoParams,
         [bool]
         $AutoUpgrade = $false
     )
     Write-Verbose -Message 'Start Set-TargetResource'
+    $env:Path = [Environment]::GetEnvironmentVariable('Path','Machine')
+    Write-Verbose -Message "Path variables: $env:Path"
 
     if (-Not (Test-ChocoInstalled)) {
         throw "cChocoPackageInstall requires Chocolatey to be installed, consider using cChocoInstaller with 'dependson' in dsc config"
     }
 
-    $isInstalled = IsPackageInstalled -pName $Name
+    if ($Version) {
+        $isInstalled = $true
+        $Version | ForEach-Object {$isInstalled = $isInstalled -and $(IsPackageInstalled -pName $Name -pVersion $_)}
+    } else {
+        $isInstalled = IsPackageInstalled -pName $Name
+    }
+
+    $isInstalledType = IsPackageInstalled -pName $Name

```

Comment:
Can you explain this line? You already know if the package status at this point (as it's checked above) so why check it again?